src/core/thread/lowpan.cpp

```diff
@@ -194,7 +195,7 @@ int Lowpan::CompressMulticast(const Ip6::Address &aIpAddr, uint16_t &aHcCtl, uin
                 memcpy(cur + 1, aIpAddr.mFields.m8 + 13, 3);
                 cur += 4;
             }
-            else if (i >= 9)
+            else if (i >= 11)
```

Comment:
minor: Is there a way to make these IPv6 address offsets less magic?  
Even a short, targeted comment to align the compression states to the RFC text would be valued.